controller/src/drone_state.rs

```diff
@@ -0,0 +1,297 @@
+                }),
+            }),
+        }],
+        DroneStateUpdate::DroneStatusMessage(msg) => vec![
+            WorldStateMessage {
+                cluster: msg.cluster.clone(),
+                message: ClusterStateMessage::DroneMessage(DroneMessage {
+                    drone: msg.drone_id.clone(),
+                    message: DroneMessageType::State {
+                        state: msg.state,
+                        timestamp,
+                    },
+                }),
+            },
+            WorldStateMessage {
+                cluster: msg.cluster.clone(),
+                message: ClusterStateMessage::DroneMessage(DroneMessage {
+                    drone: msg.drone_id.clone(),
+                    message: DroneMessageType::KeepAlive { timestamp },
+                }),
+            },
+        ],
+        DroneStateUpdate::BackendStateMessage(msg) => vec![
+            WorldStateMessage {
+                cluster: msg.cluster.clone(),
+                message: ClusterStateMessage::BackendMessage(BackendMessage {
+                    backend: msg.backend.clone(),
+                    message: BackendMessageType::State {
+                        state: msg.state,
+                        timestamp,
+                    },
+                }),
+            },
+            WorldStateMessage {
+                cluster: msg.cluster.clone(),
+                message: ClusterStateMessage::BackendMessage(BackendMessage {
+                    backend: msg.backend.clone(),
+                    message: BackendMessageType::Assignment {
+                        drone: msg.drone.clone(),
+                    },
+                }),
+            },
+        ],
+    }
+}
+
+pub async fn apply_state_message(
+    nats: &TypedNats,
+    message: &WorldStateMessage,
+) -> Result<Option<u64>> {
+    let overwrite = message.overwrite();
+    tracing::info!(?message, ?overwrite, "Publishing state message.");
+    if overwrite {
+        nats.publish_jetstream(message).await.map(Some)
+    } else {
+        nats.publish_jetstream_if_subject_empty(message).await
+    }
+}
+
+pub async fn monitor_drone_state(nats: TypedNats) -> NeverResult {
+    let mut acme_sub = nats
+        .subscribe(DroneStateUpdate::subscribe_subject_acme())
+        .await?;
+    tracing::info!("Subscribed to ACME DNS messages.");
+
+    let mut connect_sub = nats
+        .subscribe(DroneStateUpdate::subscribe_subject_connect())
+        .await?;
+    tracing::info!("Subscribed to drone connect messages.");
+
+    let mut drone_status_sub = nats
+        .subscribe(DroneStateUpdate::subscribe_subject_drone_status())
+        .await?;
+    tracing::info!("Subscribed to drone status messages.");
+
+    let mut backend_status_sub = nats
+        .subscribe(DroneStateUpdate::subscribe_subject_backend_status())
+        .await?;
+    tracing::info!("Subscribed to backend status messages.");
```

Comment:
Proposal: maybe we use "status" and "state" to mean more specific things. E.g. we could use "status" to describe heartbeat-like messages and something like "state change" for messages reporting when a backend's state has changed. Thoughts?